api/turing/service/router_version_service.go

```diff
@@ -113,7 +113,9 @@ func (service *routerVersionsService) ListRouterVersionsWithStatus(
 func (service *routerVersionsService) Save(routerVersion *models.RouterVersion) (*models.RouterVersion, error) {
 	var err error
 	tx := service.db.Begin()
-	if service.db.NewRecord(routerVersion) {
+	// For create vs update, there is no default gorm API. Upsert will not help for
+	// foreign key associations (enricher / ensembler). So, we naively compare ID==0.
+	if routerVersion.ID == 0 {
```

Comment:
Can we leave a comment mentioning what `ID == 0` means? i.e. when this router does not exist or something of the like? 😅